run_tests.py.in

```diff
@@ -315,210 +302,183 @@ def run_pack(tests_pack):  # type: (list[str]) -> None
     # Set locale to use in sort functions below
     locale.setlocale(locale.LC_ALL, 'C')
 
-    # Pattern compilation in a loop is not a good idea.
-    # Thus, we prepare patterns once.
-    re2c_pattern = re.compile(r'.*re2c[ \t]*(.*)$')
-    type_header_pattern1 = re.compile(r'--type-header[ \t]*')
-    type_header_pattern2 = re.compile(
-        r'.*--type-header[ \t]*([^ \t]*).*'
+    # remove prefix
+    outx = test[len(os.path.basename(_ctx.base_path))+1:]
+
+    os.chdir(_ctx.base_path)
+
+    with open(outx, mode='rb') as f:
+        head = f.readline().decode('utf-8').strip()
+        # generate file extension (.c for C/C++, .go for Go)
+        ext = 'go' if 're2go' in head else 'c'
+
+    outy = f'{os.path.splitext(outx)[0]}.{ext}'
+
+    # Prepare re2c flags
+    switches = head.replace('re2go', 're2c --lang go')
+    switches = re2c_pattern.sub(r'\1', switches)
+    switches = switches.replace('$INPUT', outx)
+    switches = switches.replace('$OUTPUT', outy)
+    switches = switches.replace('$DEPFILE', f'{outy}.d')
+    switches = type_header_pattern1.sub(
```

Comment:
I'm wonder if we can merge `type_header_pattern1 ` and `type_header_pattern2` and use the same pattern in bith cases. Did you try optimise this?

Reply:
I'd rather not do it in this PR, as it's unrelated.